### PR TITLE
LPD-20684 - Fix flaky fields tests

### PR DESCRIPTION
**Pull request by juanjofgliferay**
## Task
https://liferay.atlassian.net/browse/LPD-20684

## Changes
- Create fixture to ensure Data Set schema is created before running tests
- Refactor API endpoints to create Data Set related data
- Remove old `Fields` related files

## Tests
Screenshot from 2024-04-09 10-49-30

Screenshot from 2024-04-09 11-03-10

### CODE REVIEW

**Comment by juanjofgliferay** [comment]
This PR still needs https://github.com/liferay-frontend/liferay-portal/pull/4113 & https://github.com/liferay-frontend/liferay-portal/pull/4114 (feature flag removal)

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Review by markocikos — APPROVED** [review]
@juanjofgliferay LGTM! A non-blocking nit inline. I would only wait until https://github.com/brianchandotcom/liferay-portal/pull/148837 is synced, and rebase to it, just to make sure there are no conflicts.

File: modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts [hunk 1]
```diff
@@ -42,196 +53,514 @@ test.afterEach(async ({dataSetManagerApiHelpers}) => {
...
 });
 
-test('Configure cards visualization mode @LPD-10735', async ({
-	visualizationModesPage,
-}) => {
-	await test.step('Navigate to cards visualization mode page', async () => {
-		await visualizationModesPage.goto({
-			dataSetLabel,
-			viewLabel,
+test.describe('Visualization Modes in Data Set Manager', () => {
```
[the remark this answers is not in the document]
Comment (juanjofgliferay) [reply modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts:56]: It's all about grouping. We internally identify `DSM` vs `fragment` tests, so it I think is clear enough without `describe`.

[the remark this answers is not in the document]
Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts:56]: Oh, I see. Would it maybe make sense to move fragment tests to their own file? I originally thought of doing them as a part of configure in DSM + check in fragment, all in one `test`. But, if they are isolated and generate their own sample data, then moving to own file makes more sense.

**Comment by juanjofgliferay** [comment]
Data Set tests :rocket: 
https://testray.liferay.com/reports/production/logs/2024-04/test-1-39/test-portal-acceptance-pullrequest(master)/9855/playwright-js-tomcat90-mysql57-jdk8/6/0/playwright-report/

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/action/SaveFDSFieldsMVCResourceCommand.java [hunk 2]
```diff
@@ -84,9 +87,9 @@ protected void doTransactionalCommand(
 				).put(
 					"renderer", "default"
 				).put(
-					"sortable", true
+					"sortable", !StringUtil.equals(type, "object")
```
[the remark this answers is not in the document]
Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/action/SaveFDSFieldsMVCResourceCommand.java]: Nothing to worry about.

**Comment by juanjofgliferay** [comment]
ci:test:sf

File: modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (juanjofgliferay) [reply modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts:56]: @markocikos Any suggestion for the naming? My first thought was to use a prefix to identify the type of test `visualizationModes.spec.ts` & `fragmentVisualizationModes.spec.ts`. 
However, this approach will have the drawback of alphabetical order, so we will end up with a mix of DSM and fragment tests.

```
- creationActions.spec.ts
- dataSets.spec.ts
- fragmentCreationActions.spec.ts
- fragmentDataSets.spec.ts
- fragmentItemActions.spec.ts
- fragmentSettings.spec.ts
- fragmentVisualizationModes.spec.ts
- itemActions.spec.ts
- settings.spec.ts
- visualizationModes.spec.ts
```

Another option could be using the name file name but using a folder to contain each case:
```
|_/dsm
    |- creationActions.spec.ts
    |- dataSets.spec.ts
    |- itemActions.spec.ts
    |- settings.spec.ts
    |- visualizationModes.spec.ts   
|_/fragment
    |- creationActions.spec.ts
    |- dataSets.spec.ts
    |- itemActions.spec.ts
    |- settings.spec.ts
    |- visualizationModes.spec.ts   

```
What do you think? cc/ @chestofwonders @dsanz

[the remark this answers is not in the document]
Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts:56]: We can do this in a followup, or as a part of this PR, as you wish. As far as I'm concerned, this PR is good to go :rocket:

**Comment by juanjofgliferay** [comment]
ci:forward

**Pull request closed without merging**